lib/babel/transformation/transformers/spec/function-name.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+var t           = require("../../../types");
+var util        = require("../../../util");
+
+var propertyFunctionVisitor = {
+  enter: function (node, parent, scope, state, file) {
+    if (t.isReferencedIdentifier(node, parent, { name: state.name }) && scope.getBindingIdentifier(node.name) === state.binding) {
+      return state.getOuter();
+    }
+  }
+};
+
+exports.FunctionExpression = function (node, parent, scope, file) {
+  if (node.id) return;
+  var id;
+  if (t.isProperty(parent)) {
+    id = parent.key;
+  } else if (t.isVariableDeclarator(parent)) {
+    id = parent.id;
+  } else {
+    return;
+  }
+  var binding = scope.getBindingIdentifier(id.name);
+  var outerId, selfGlobalId;
+  scope.traverse(node, propertyFunctionVisitor, {
+    name: id.name,
+    binding: binding,
+
+    getOuter: function () {
+      if (!binding) {
+        return t.memberExpression(
+          selfGlobalId || (selfGlobalId = file.addHelper("self-global")),

```

Comment:
But if you wish, this can be easily removed in favor of `_getOuter()` everywhere.